pyro/infer/autoguide/guides.py

```diff
@@ -250,6 +269,12 @@ def __init__(self, model, prefix="auto", init_loc_fn=init_to_median):
         model = InitMessenger(self._init_loc_fn)(model)
         super().__init__(model, prefix=prefix)
 
+    def _init_params(self):
```

Comment:
Should we use the pattern
```py
def _setup_prototype(self, *args, **kwars):
    super()._setup_prototype(*args, **kwargs)
    for name, site in ...:
        # set up parameters
```
as is used in `AutoContinuous` below?

Reply:
Yes, I think that sounds reasonable. Let me address this.